### ISSUE TITLE
Kiali Sample Addon throws exception

### ISSUE THREAD

**Issue by sumitdhameja** [kiali/kiali#5328]
I have been trying to install kiali using istio sample addons and have been receiving the below exception as soon as I bring up Kiali UI.
Kubernetes Version is 1.20.
istio - 1.13.2
kiali - v1.45.0
Any help to debug the issue would be helpful. 

```
kiali-688cfc84dd-jssm7 2022/07/21 20:01:05 maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined
kiali-688cfc84dd-jssm7 2022-07-21T20:01:05Z INF Kiali: Version: v1.45.1, Commit: 3fd79452338f86077bbca23e0ada033df9d12aed
kiali-688cfc84dd-jssm7 
kiali-688cfc84dd-jssm7 2022-07-21T20:01:05Z INF Using authentication strategy [token]
kiali-688cfc84dd-jssm7 2022-07-21T20:01:05Z INF Kiali: Console version: 1.45.1
kiali-688cfc84dd-jssm7 2022-07-21T20:01:05Z INF Server endpoint will start at [:20001/kiali]
kiali-688cfc84dd-jssm7 2022-07-21T20:01:05Z INF Server endpoint will serve static content from [/opt/kiali/console]
kiali-688cfc84dd-jssm7 2022-07-21T20:01:05Z INF Starting Metrics Server on [:9090]
kiali-688cfc84dd-5jdqf 2022/07/21 20:01:54 maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined
kiali-688cfc84dd-5jdqf 2022-07-21T20:01:54Z INF Kiali: Version: v1.45.1, Commit: 3fd79452338f86077bbca23e0ada033df9d12aed
kiali-688cfc84dd-5jdqf 
kiali-688cfc84dd-5jdqf 2022-07-21T20:01:54Z INF Using authentication strategy [token]
kiali-688cfc84dd-5jdqf 2022-07-21T20:01:54Z INF Kiali: Console version: 1.45.1
kiali-688cfc84dd-5jdqf 2022-07-21T20:01:54Z INF Server endpoint will start at [:20001/kiali]
kiali-688cfc84dd-5jdqf 2022-07-21T20:01:54Z INF Server endpoint will serve static content from [/opt/kiali/console]
kiali-688cfc84dd-5jdqf 2022-07-21T20:01:54Z INF Starting Metrics Server on [:9090]
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:16Z INF Kiali Cache is active for namespaces [.*]
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:16Z INF [Prom Cache] Enabled
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z INF Adding a RegistryRefreshHandler for [istio-system]
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z INF Waiting for Kiali cache for [namespace: istio-system] to sync
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z INF Kiali cache for [namespace: istio-system] started
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z ERR grafana URL is not set in Kiali configuration
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z INF Adding a RegistryRefreshHandler for [test-namespace]
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z INF Waiting for Kiali cache for [namespace: test-namespace] to sync
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:17Z INF Kiali cache for [namespace: test-namespace] started
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z ERR Workload ************************** is not found as Job
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z ERR Workload ************************** is not found as Job
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z ERR Workload ************************** is not found as Job
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z ERR Workload ************************** is not found as Job
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z INF Create Jaeger GRPC client tracing.istio-system:16685
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z ERR Workload ************************** is not found as ReplicaSet
kiali-688cfc84dd-5jdqf 2022-07-21T20:04:38Z ERR Workload ************************** is not found as ReplicaSet
kiali-688cfc84dd-5jdqf 2022/07/21 20:04:39 http: panic serving 127.0.0.1:33046: runtime error: invalid memory address or nil pointer dereference
kiali-688cfc84dd-5jdqf goroutine 403 [running]:
kiali-688cfc84dd-5jdqf net/http.(*conn).serve.func1(0xc00069e320)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:1824 +0x153
kiali-688cfc84dd-5jdqf panic(0x1cfce40, 0x2f8b350)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/runtime/panic.go:971 +0x499
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).getClusterId(0xc00091d148, 0x1fbbc61, 0xc00033a360)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:282 +0xdf
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).buildRegistryServices(0xc00091d148, 0x2fe5740, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:311 +0x54
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).buildServiceList(0xc00091d148, 0xc0006c838a, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc003748280, 0x1, ...)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:195 +0x2e5
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).GetServiceList(0xc00091d148, 0xc0006c838a, 0xc, 0x100, 0xc000808000, 0x14a, 0x5, 0x8, 0x0)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:163 +0x5bb
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.fetchNamespaceApps(0xc00091cfc0, 0xc0006c838a, 0xc, 0x0, 0x0, 0x1ab88a8, 0x1e2ad40, 0xc0010f1a10)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/apps.go:273 +0x22c
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*HealthService).GetNamespaceAppHealth(0xc00091cfe8, 0xc0006c838a, 0xc, 0xc0006c83b4, 0x3, 0xc0ae8a3056e26b69, 0x215ff52666, 0x2fb57a0, 0xc0006c83b4, 0x3, ...)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/health.go:103 +0x52
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/handlers.NamespaceHealth(0x2255bf0, 0xc001094c48, 0xc000b28900)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/handlers/health.go:42 +0x1ee
kiali-688cfc84dd-5jdqf net/http.HandlerFunc.ServeHTTP(0x2089440, 0x2255bf0, 0xc001094c48, 0xc000b28900)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2069 +0x44
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/routing.metricHandler.func1(0x2262c70, 0xc0004d0230, 0xc000b28900)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/routing/router.go:141 +0x11e
kiali-688cfc84dd-5jdqf net/http.HandlerFunc.ServeHTTP(0xc0004b9320, 0x2262c70, 0xc0004d0230, 0xc000b28900)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2069 +0x44
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/handlers.AuthenticationHandler.Handle.func1(0x2262c70, 0xc0004d0230, 0xc000a8ed00)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/handlers/authentication.go:688 +0x1f7
kiali-688cfc84dd-5jdqf net/http.HandlerFunc.ServeHTTP(0xc0003acd50, 0x2262c70, 0xc0004d0230, 0xc000a8ed00)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2069 +0x44
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/server.plainHttpMiddleware.func1(0x2262c70, 0xc0004d0230, 0xc000a8ed00)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/server/server.go:127 +0x6c
kiali-688cfc84dd-5jdqf net/http.HandlerFunc.ServeHTTP(0xc001466528, 0x2262c70, 0xc0004d0230, 0xc000a8ed00)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2069 +0x44
kiali-688cfc84dd-5jdqf github.com/gorilla/mux.(*Router).ServeHTTP(0xc0002aafc0, 0x2262c70, 0xc0004d0230, 0xc000a8e600)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/pkg/mod/github.com/gorilla/mux@v1.7.4/mux.go:210 +0xd3
kiali-688cfc84dd-5jdqf github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1(0x2262b80, 0xc000c94540, 0xc000a8e600)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/pkg/mod/github.com/!n!y!times/gziphandler@v1.1.1/gzip.go:336 +0x22d
kiali-688cfc84dd-5jdqf net/http.HandlerFunc.ServeHTTP(0xc0007b5500, 0x2262b80, 0xc000c94540, 0xc000a8e600)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2069 +0x44
kiali-688cfc84dd-5jdqf net/http.(*ServeMux).ServeHTTP(0xc000143c40, 0x2262b80, 0xc000c94540, 0xc000a8e600)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2448 +0x1ad
kiali-688cfc84dd-5jdqf net/http.serverHandler.ServeHTTP(0xc0007de000, 0x2262b80, 0xc000c94540, 0xc000a8e600)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:2887 +0xa3
kiali-688cfc84dd-5jdqf net/http.(*conn).serve(0xc00069e320, 0x2266b58, 0xc000a9e000)
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:1952 +0x8cd
kiali-688cfc84dd-5jdqf created by net/http.(*Server).Serve
kiali-688cfc84dd-5jdqf     /usr/local/go/src/net/http/server.go:3013 +0x39b
kiali-688cfc84dd-5jdqf panic: runtime error: invalid memory address or nil pointer dereference
kiali-688cfc84dd-5jdqf [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x1a3ea7f]
kiali-688cfc84dd-5jdqf 
kiali-688cfc84dd-5jdqf goroutine 1838 [running]:
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).getClusterId(0xc006fce608, 0x1fbbc61, 0xc000b14fc0)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:282 +0xdf
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).buildRegistryServices(0xc006fce608, 0xc003df2600, 0x32, 0x40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:311 +0x54
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).buildServiceList(0xc006fce608, 0xc002fee31a, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc000321900, 0x7, ...)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:195 +0x2e5
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*SvcService).GetServiceList(0xc006fce608, 0xc002fee31a, 0xc, 0x0, 0x0, 0x0, 0x0, 0x200030000, 0x300000002)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/services.go:163 +0x5bb
kiali-688cfc84dd-5jdqf github.com/kiali/kiali/business.(*IstioValidationsService).fetchServices(0xc006fce660, 0xc0042ccde0, 0xc002fee31a, 0xc, 0xc0042ccba0, 0xc002f8a9e4)
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/istio_validations.go:254 +0x112
kiali-688cfc84dd-5jdqf created by github.com/kiali/kiali/business.(*IstioValidationsService).GetValidations
kiali-688cfc84dd-5jdqf     /home/fedora/workspace/kiali-upstream/kiali-release/src/github.com/kiali/kiali/business/istio_validations.go:74 +0x505
Stream closed EOF for istio-system/kiali-688cfc84dd-5jdqf (kiali)

```

**Comment by jmazzitelli** [comment]
This sounds familiar. It could be this - fixed in Kiali v1.48:

* https://github.com/kiali/kiali/issues/4788
* https://github.com/kiali/kiali/pull/4791

Looks like it is caused by a misconfiguration (based on reading the issue and PR).

To fix, either upgrade Kiali to 1.48+ or configure Kiali with `external_services.istio.istiod_deployment_name` so it matches the actual deployment name (by default `istiod`) -- if it doesn't match, that panic occurs (until 1.48).

**Comment by sumitdhameja** [comment]
Awesome! Thank you configuring Kiali config worked
